src/main/java/hudson/remoting/Launcher.java

```diff
@@ -227,6 +245,12 @@ public void run() throws Exception {
             if (this.noReconnect) {
                 jnlpArgs.add("-noreconnect");
             }
+            if (candidateCertificates != null && !candidateCertificates.isEmpty()) {
+                for (String c: candidateCertificates) {
+                    jnlpArgs.add("-cert");
+                    jnlpArgs.add(c);
```

Comment:
I've thought that it maps only comma-separated params. Have you tested the multiple certificate case?